### PR TITLE
Fix parsing kubectl output with warnings

### PR DESCRIPTION
**Pull request by usmonster**
Exclude stderr (e.g. deprecation warnings) when parsing command output.

**What this PR does / why we need it**: It ensures that warnings or other output that goes to stderr won't be parsed when trying to list deployments, pods, etc. Today, this is not the case, and many  `ct` subcommands will fail in such situations (e.g. when the [gke-gcloud-auth-plugin](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke) isn't yet installed).

**Special notes for your reviewer**:
It's possible that the original implementation wanted to include stderr in the output for these commands for a reason, but this feels like a simple oversight.

It's also possible that there are other similar places in the code that require the same modification, so please feel free to update the PR or request these changes, which I will then apply. Thanks!

### CODE REVIEW

**Comment by usmonster** [comment]
Bump! @cpanato, @smlx—as recent committers on this file, can you please take a look as soon as you get a chance? 🙏 Thanks!

**Comment by usmonster** [comment]
Hello, just checking in—@davidkarlsen, @scottrigby, do you think you can take a quick look at this fix and hopefully get it into the next version? Thanks again! 🙇

**Comment by davidkarlsen** [comment]
I'm a bit uneasy to merge it straight as I suspect some unexpected side-effects, but I'd like @unguiculus comments as the original author. Do you remember the reasoning between getting both streams here?

**Comment by unguiculus** [comment]
Maybe we should adjust `RunProcessAndCaptureOutput` so that it also logs the proc's stderr to stderr, so we don't swallow any errors. Otherwise, I'd be fine with this change.

**Comment by usmonster** [comment]
Thanks @unguiculus! I like your suggestion as a separate PR, which could also help with #359.

Regarding @davidkarlsen's concerns, there should be no unwanted side effects since we're now only filtering stderr output that isn't currently used or expected in these cases.

Please feel free to merge if there's nothing else to change, and thanks again for the review!

**Comment by usmonster** [comment]
Just a friendly bump to @davidkarlsen & @scottrigby. 🙂

Do you need anything else from me before this can be merged? Thanks again!

**Review by davidkarlsen — APPROVED** [review]
lgtm

**Review by cpanato — APPROVED** [review]

**Pull request merged**
